### PR TITLE
Fixed a double precision floating number problem

### PR DESCRIPTION
**Pull request by tong-hao**
### What changes were proposed in this pull request?
Fixed a double precision floating number problem for #1293

### Why are the changes needed?
The default double precision in c library format  is 6 bits decimal.

### Does this PR introduce any user-facing change?
Yes.
```
(user@127.0.0.1:3699) [idmapping]> yield 1.01234567890123456789
=====================
| 1.012345678901235 |
=====================
| 1.012345678901235 |
---------------------
Got 1 rows (Time spent: 264/1204 us)


(user@127.0.0.1:3699) [test3]> fetch prop on num 1;
=============================
| num.i | num.d             |
=============================
| 1     | 1.100000000000000 |
-----------------------------
Got 1 rows (Time spent: 16.013/16.962 ms)


```

### How was this patch tested?
Manual testing

### CODE REVIEW

**Comment by panda-sheep** [comment]
Excellent！

File: src/common/filter/Expressions.cpp [hunk 1]
```diff
@@ -308,8 +308,12 @@ std::string PrimaryExpression::toString() const {
         case VAR_INT64:
             snprintf(buf, sizeof(buf), "%ld", boost::get<int64_t>(operand_));
             break;
-        case VAR_DOUBLE:
-            return std::to_string(boost::get<double>(operand_));
+        case VAR_DOUBLE: {
+            int digits10 = std::numeric_limits<double>::digits10;
+            const char *fmt = folly::sformat("%.{}lf", digits10).c_str();
+            snprintf(buf, sizeof(buf), fmt, boost::get<double>(operand_));
```
Comment (dutor) [inline src/common/filter/Expressions.cpp]: `fmt` will be a dangling pointer here.

File: src/console/CmdProcessor.cpp [hunk 2]
```diff
@@ -89,18 +89,22 @@ void CmdProcessor::calColumnWidths(
                     break;
                 }
                 case cpp2::ColumnValue::Type::single_precision: {
-                    GET_VALUE_WIDTH(float, single_precision, "%f")
+                    int digits10 = std::numeric_limits<float>::digits10;
+                    const char *fmtValue = folly::sformat("%.{}f", digits10).c_str();
+                    GET_VALUE_WIDTH(float, single_precision, fmtValue);
```
Comment (dutor) [inline src/console/CmdProcessor.cpp]: The same to `fmtValue`

File: src/console/CmdProcessor.cpp [hunk 3]
```diff
@@ -89,18 +89,22 @@ void CmdProcessor::calColumnWidths(
...
                     if (genFmt) {
-                        formats[idx] =
-                            folly::stringPrintf(" %%-%ldf |", widths[idx]);
+                        const char *fmt = folly::sformat(" %%-%ld.{}f |", digits10).c_str();
+                        formats[idx] = folly::stringPrintf(fmt, widths[idx]);
```
Comment (dutor) [inline src/console/CmdProcessor.cpp]: ditto

File: src/console/CmdProcessor.cpp [hunk 4]
```diff
@@ -89,18 +89,22 @@ void CmdProcessor::calColumnWidths(
...
                     }
                     break;
                 }
                 case cpp2::ColumnValue::Type::double_precision: {
-                    GET_VALUE_WIDTH(double, double_precision, "%lf")
+                    int digits10 = std::numeric_limits<double>::digits10;
+                    const char *fmtValue = folly::sformat("%.{}lf", digits10).c_str();
+                    GET_VALUE_WIDTH(double, double_precision, fmtValue);
```
Comment (dutor) [inline src/console/CmdProcessor.cpp]: ditto.

File: src/console/CmdProcessor.cpp [hunk 5]
```diff
@@ -89,18 +89,22 @@ void CmdProcessor::calColumnWidths(
...
                     if (genFmt) {
-                        formats[idx] =
-                            folly::stringPrintf(" %%-%ldlf |", widths[idx]);
+                        const char *fmt = folly::sformat(" %%-%ld.{}lf |", digits10).c_str();
+                        formats[idx] = folly::stringPrintf(fmt, widths[idx]);
```
Comment (dutor) [inline src/console/CmdProcessor.cpp]: ditto.

Comment (tong-hao) [reply src/console/CmdProcessor.cpp]: Thanks for your reviews.  I fixed it.

**Comment by dutor** [comment]
@tong-hao There seems some testing failures in `yield_test`. Would you please address those?

**Pull request merged**